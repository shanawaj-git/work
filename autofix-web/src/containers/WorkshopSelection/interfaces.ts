import { LabelType } from "@/services/common";
import { Location } from "@/Types/woosmap";

export interface WorkshopSelectionType {
  en?: { labels: LabelType[] };
  ar?: { labels: LabelType[] };
}

export interface LatLng {
  lat: () => number;
  lng: () => number;
}

export interface workshopType {
  id: string;
  name: string;
  rating: number;
  address: {
    location: {
      coordinates: number[];
    };
  };
}

export interface GetDirections {
  getNorthEast: () => LatLng;
  getSouthWest: () => LatLng;
}

export interface MapObjType {
  getBounds: () => GetDirections;
  getZoom: () => number;
  panTo: (e: LatLngType) => void;
  addListener(arg0: string, arg1: () => void): unknown;
}

export interface MapType {
  map: MapObjType;
  mapWindowRef: {};
}
export interface LatLngType {
  lat: number;
  lng: number;
}

export interface LocationType {
  type: string;
  coordinates: number[];
}

export interface BoundPayloadType {
  locationBoxBounds: {
    bottomLeft: LocationType;
    topRight: LocationType;
  };
}

interface GeoLocation {
  latitude: number;
  longitude: number;
}

export interface WorkshopsPayload {
  payload: {
    locationBoxBounds: {
      bottomLeft: GeoLocation;
      topRight: GeoLocation;
    };
    limit: number;
  };
}

export interface HeaderChildrenProps {
  searchText: string;
  setSearchText: Function;
  onAddressSelection: (location: Location) => void;
}

export interface WorkshopMarkerProps {
  name: string;
  rating: number;
}
