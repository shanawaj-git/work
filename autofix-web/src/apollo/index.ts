import {
  ApolloClient,
  ApolloLink,
  HttpLink,
  InMemoryCache,
  concat,
} from "@apollo/client";
import fetch from "cross-fetch";

interface GetClientType {
  token?: undefined | string;
  link: string;
}
const getClient = ({ token = undefined, link }: GetClientType) => {
  const httpLink = new HttpLink({
    fetch,
    uri: link,
  });

  const authMiddleware = new ApolloLink((operation, forward) => {
    return forward(operation);
  });

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: concat(authMiddleware, httpLink),
  });
};

const client = getClient({
  link: `${process.env.NEXT_PUBLIC_STRAPI_SERVICE}/graphql`,
});

export default client;

export const autoFixClient = getClient({
  link: `${process.env.NEXT_PUBLIC_AUTOFIX_SERVICE}/graphql`,
});
