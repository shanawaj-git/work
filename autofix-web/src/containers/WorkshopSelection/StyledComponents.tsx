import styled from "styled-components";
import TextField from "@mui/material/TextField";
import get from "lodash.get";
import { utils } from "@albathanext/design-system";

export const StyledTextField = styled(TextField)`
  width: 100%;
  border-radius: 4px;
  background-color: #1f2222;
  padding-right: 0px;

  .Mui-focused .MuiOutlinedInput-notchedOutline {
    border: 0px solid ${utils.colorSelector("c100")};
  }

  & .MuiInputBase-formControl {
    padding-right: 30px !important;
  }

  .MuiAutocomplete-inputRoot {
    border-radius: ${(props) =>
      get(props, "theme.overrides.button.borderRadius", "6px")};
    border-color: ${utils.colorSelector("e101")};
    color: white;
    padding-left: 50px;
    padding-right: 0px;
  }

  .MuiAutocomplete-popupIndicator {
    display: none;
  }
`;
