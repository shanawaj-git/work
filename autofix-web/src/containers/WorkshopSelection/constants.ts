export const initialCoordinates =
  process.env.NEXT_PUBLIC_INITIAL_COORDINATES &&
  Object.values(JSON.parse(process.env.NEXT_PUBLIC_INITIAL_COORDINATES));

export const ZOOM_LEVELS = {
  levelToHideAll: 8,
  levelToDisplayAll: 10,
};

export const mapProps = {
  initialCoordinates,
  woosMapKey: process.env.NEXT_PUBLIC_WOOSMAP_KEY,
  googlePlacesKey: process.env.NEXT_PUBLIC_GOOGLE_PLACES_KEY,
  googleMapKey: process.env.NEXT_PUBLIC_GOOGLE_KEY,
};
