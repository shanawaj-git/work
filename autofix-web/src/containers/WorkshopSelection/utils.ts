import {
  BoundPayloadType,
  GetDirections,
  LatLngType,
  MapObjType,
} from "./interfaces";

export const markerIcon = (icon: string) => ({
  scaledSize: {
    height: 24,
    width: 24,
  },
  url: icon,
});

export const getBoundLocation = ({
  map,
  onBoundChange,
}: {
  map: MapObjType;
  onBoundChange: (e: BoundPayloadType) => void;
}) => {
  const bounds = map.getBounds();
  const southWest = getSouthWestCoordinates(bounds);
  const northEast = getNorthEastCoordinates(bounds);
  const result = buildBoundPayload(northEast, southWest);
  onBoundChange(result);
};

const getSouthWestCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getSouthWest().lat(),
  lng: bounds.getSouthWest().lng(),
});

const getNorthEastCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getNorthEast().lat(),
  lng: bounds.getNorthEast().lng(),
});

const buildBoundPayload = (
  northEast: LatLngType,
  southWest: LatLngType
): BoundPayloadType => ({
  locationBoxBounds: {
    bottomLeft: {
      type: "Point",
      coordinates: [southWest.lng, southWest.lat],
    },
    topRight: { type: "Point", coordinates: [northEast.lng, northEast.lat] },
  },
});
