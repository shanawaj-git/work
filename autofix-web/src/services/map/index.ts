import {
  autocompleteMultiResponse,
  detailsMultiType,
  getPlaceDetailsType,
  multiSearchType,
  processAutoCompleteType,
  windowMultiSearch,
} from "src/Types/woosmap";

export const MULTI_SEARCH_CDN =
  "https://sdk.woosmap.com/multisearch/multisearch.js?version=6";
const UAE_COUNTRY_CODE = "AE";
const components = {
  country: [UAE_COUNTRY_CODE],
};

// Get the detailed
export const getPlaceDetails = (
  detailsMulti: (e: getPlaceDetailsType) => Promise<detailsMultiType>,
  obj: getPlaceDetailsType
) => detailsMulti(obj);

// Search By Text
export const getPlacesByName = (
  multiSearch: Function,
  value: string
): Promise<autocompleteMultiResponse[]> => multiSearch(value);

export const processAutoComplete = (
  searchResults: autocompleteMultiResponse[]
): processAutoCompleteType[] =>
  searchResults.map((searchResultItem) => {
    const { id, api } = searchResultItem;
    const { terms = [], description } = searchResultItem?.item;
    return {
      addressComponents: terms.map((term) => term.value),
      label: description,
      id: id,
      api,
    };
  });

export const getWoosMap = () =>
  (window as unknown as windowMultiSearch).woosmap;

export const initMultiSearchAPI = (): multiSearchType =>
  getWoosMap().multisearch({
    apiOrder: ["places", "localities", "address"],
    components,
    localities: {
      key: `${process.env.NEXT_PUBLIC_WOOS_MAP_KEY}`,
      params: {
        language: "en",
        components,
      },
    },
    address: {
      key: `${process.env.NEXT_PUBLIC_WOOS_MAP_KEY}`,
      params: {
        language: "en",
        components,
      },
    },
    places: {
      params: {
        language: "en",
        components,
      },
      key: `${process.env.NEXT_PUBLIC_GOOGLE_PLACES_KEY}`,
    },
  });
