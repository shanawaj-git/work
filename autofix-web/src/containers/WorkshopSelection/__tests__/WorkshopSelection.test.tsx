import React, { useState } from "react";
import { act, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import WorkshopSelection, { pageName } from "../index";
import { withWrapper } from "@/utils/IntlProviderWrapper";
import * as Map from "@/services/map";
import * as designSystem from "@albathanext/design-system";
import * as Utils from "@/utils/index";
import * as containerUtils from "../utils";
import { autoFixClient } from "@/apollo/index";
import { ZOOM_LEVELS } from "../constants";

const workshops = [
  {
    __typename: "WorkshopResource",
    id: "eda51a30-cf21-4865-97b9-36bcc32f2850",
    name: "Tanya Jast",
    rating: 2.4,
    address: {
      __typename: "Address",
      location: {
        __typename: "Geolocation",
        coordinates: [54.48576885812361, 24.70185728066236],
      },
    },
  },
  {
    __typename: "WorkshopResource",
    id: "ca7853b7-61b1-4091-a90a-fb6c3923e928",
    name: "Silvia Abernathy",
    rating: 4.5,
    address: {
      __typename: "Address",
      location: {
        __typename: "Geolocation",
        coordinates: [54.63820837706659, 24.5548710433063],
      },
    },
  },
  {
    __typename: "WorkshopResource",
    id: "28e3e091-61c8-48ce-b42c-575487431f54",
    name: "Rufus Jenkins",
    rating: 4.2,
    address: {
      __typename: "Address",
      location: {
        __typename: "Geolocation",
        coordinates: [54.57471993370657, 24.479657108116715],
      },
    },
  },
  {
    __typename: "WorkshopResource",
    id: "be7d2e8a-24fa-4dbd-a02b-49f1819b87a8",
    name: "Chad Bashirian",
    rating: 2.6,
    address: {
      __typename: "Address",
      location: {
        __typename: "Geolocation",
        coordinates: [54.5854845372338, 24.851858669019506],
      },
    },
  },
];

/**
 * Criteria :
 * Component should render
 * Component should contain map component
 * Component should contain input
 * On Input change, Should trigger API
 * Map Bound Change, Should trigger API
 *
 */

jest.mock("@/apollo/index", () => ({
  autoFixClient: {
    query: jest.fn().mockImplementation(() => ({
      data: {
        fetchWorkshops: [],
      },
    })),
  },
}));

jest.mock("@albathanext/design-system");
const mapInstance = {
  id: 10,
  mapWindowRef: {
    LatLng: jest.fn(),
    Point: jest.fn(),
  },
  map: {
    addListener: jest.fn().mockImplementation((event, callback) => {
      callback();
    }),
    getCenter: jest.fn(),
    getZoom: jest.fn(),
    panTo: jest.fn(),
    getBounds: jest.fn().mockReturnValue({
      getNorthEast: jest.fn().mockReturnValue({
        lat: jest.fn().mockReturnValue(20.222),
        lng: jest.fn().mockReturnValue(20.222),
      }),
      getSouthWest: jest.fn().mockReturnValue({
        lat: jest.fn().mockReturnValue(20.222),
        lng: jest.fn().mockReturnValue(20.222),
      }),
    }),
  },
};
jest.mock("react-dom/server", () => ({
  renderToString: jest.fn(),
}));

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = () => ({ current: mapInstance });
  return {
    ...originReact,
    useRef: mUseRef,
    useState: jest.fn().mockImplementation((e) => [e, jest.fn()]),
    useEffect: jest.fn().mockImplementationOnce((e) => e()),
  };
});

jest.mock("lodash", () => ({
  debounce: (func: Function) => (e: unknown) => func(e),
}));

const mapMockFN = jest.fn();

jest.mock("@albathanext/design-system", () => ({
  ...jest.requireActual("@albathanext/design-system"),
  useScript: jest.fn().mockReturnValue(true),
  Map: jest.fn().mockImplementation(({ onCenterChange, ...e }) => {
    mapMockFN(e);
    onCenterChange();
    return <div data-testid="map">{e.children}</div>;
  }),
  Marker: jest.fn().mockImplementation((e) => {
    return <div data-testid={"marker"}></div>;
  }),
}));

const autocompleteMulti = jest.fn();
const detailsMulti = jest.fn();

global.woosmap = {
  multisearch: jest.fn().mockReturnValue({
    autocompleteMulti,
    detailsMulti,
  }),
};

describe("<WorkshopSelection />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("Component should render", () => {
    const { container, getByTestId } = render(
      withWrapper(<WorkshopSelection />)
    );
    expect(container.innerHTML).toMatchSnapshot();
    expect(getByTestId("workshop.selection")).toBeTruthy();
    jest.restoreAllMocks();
  });

  /**
   * Should Render Map
   * OnBound Change Event Execute the side effects
   */
  test("Component should contain map component", async () => {
    useState.mockImplementation((e) => [e, jest.fn()]);
    const mockLatLng = { latitude: 12313.2123213, longitude: 12313.1 };
    const fn = jest.fn().mockImplementation(async (e) => {
      return mockLatLng;
    });
    jest.spyOn(Utils, "getCurrentLocation").mockImplementation(fn);
    autoFixClient.query.mockReturnValue({
      data: {
        fetchWorkshops: workshops,
      },
    });

    const spy = jest.spyOn(containerUtils, "getBoundLocation");
    jest.spyOn(React, "useEffect").mockImplementationOnce((e) => {
      e();
    });
    jest.spyOn(React, "useEffect").mockImplementationOnce((e) => {
      e();
    });

    render(withWrapper(<WorkshopSelection />));
    expect(mapMockFN).toBeCalled();

    const { rerender, container } = render(withWrapper(<WorkshopSelection />));
    expect(container.innerHTML).toMatchSnapshot();
    act(() => {
      rerender(withWrapper(<WorkshopSelection />));
    });

    await waitFor(
      () => {
        expect(spy).toBeCalled();
      },
      { timeout: 3000 }
    );
    expect(mapInstance.map.panTo).toBeCalled();
  });

  /**
   * Should Render Input
   * Should Inject Script
   * OnChange Event Should Call API
   */
  test("Component should contain input", async () => {
    const processAutoComplete = jest.fn().mockReturnValue([
      {
        id: "option1",
        label: "option 1",
      },
    ]);
    jest
      .spyOn(Map, "processAutoComplete")
      .mockImplementation(processAutoComplete);
    jest.spyOn(Map, "getPlacesByName").mockImplementation(jest.fn());

    const script = designSystem.useScript;

    const { getByPlaceholderText } = render(withWrapper(<WorkshopSelection />));
    const input = getByPlaceholderText("Search for an address or workshop");
    expect(input).toBeTruthy();

    expect(script).toBeCalled();
  });

  test("Component Should render markers when state update", () => {
    const state = workshops.reduce((acc, curr) => {
      acc[curr.id] = curr;
      return acc;
    }, {});

    useState.mockImplementationOnce(() => [state, jest.fn()]);
    useState.mockImplementation((e) => [e, jest.fn()]);
    const { container, getAllByTestId } = render(
      withWrapper(<WorkshopSelection />)
    );

    expect(getAllByTestId("marker").length).toBeGreaterThan(2);
    expect(container.innerHTML).toMatchSnapshot();
  });

  test("Should call getCurrentLocation on Mount and execute panTo", async () => {
    const mockLatLng = { latitude: 12313.2123213, longitude: 12313.1 };
    const fn = jest.fn().mockImplementation(async (e) => {
      return mockLatLng;
    });
    jest.spyOn(Utils, "getCurrentLocation").mockImplementation(fn);
    jest.spyOn(React, "useEffect").mockImplementationOnce((e) => {
      e();
    });
    jest.spyOn(React, "useEffect").mockImplementationOnce((e) => {
      e();
    });
    render(withWrapper(<WorkshopSelection />));
    await waitFor(() => {
      expect(fn).toBeCalled();
      expect(mapInstance.map.panTo).toHaveBeenCalled();
    });
  });

  test("it should ask for location access on mount and failed", () => {
    const fn = async () => {
      throw new Error("Error");
    };
    jest.spyOn(Utils, "getCurrentLocation").mockImplementation(fn);
    jest.spyOn(React, "useEffect").mockImplementationOnce((cb) => cb());
    render(withWrapper(<WorkshopSelection />));
    waitFor(() => {
      expect(mapInstance.map.panTo).not.toHaveBeenCalled();
    });
  });

  test("it should mount", () => {
    render(withWrapper(<WorkshopSelection />));
    const workshopSelection = screen.getByTestId(pageName);
    expect(workshopSelection).toBeInTheDocument();
  });

  test("it should hide all workshops when zoom level is equal or bigger than hide all level", () => {
    jest
      .spyOn(mapInstance.map, "getZoom")
      .mockImplementation(() => ZOOM_LEVELS.levelToHideAll);

    render(withWrapper(<WorkshopSelection />));
    waitFor(() => {
      expect(mapInstance.map.getZoom).toHaveBeenCalled();
    });

    expect(autoFixClient.query).not.toHaveBeenCalled();
  });

  test("it should hide all workshops when zoom level is equal or bigger than show all level", () => {
    jest
      .spyOn(mapInstance.map, "getZoom")
      .mockImplementation(() => ZOOM_LEVELS.levelToDisplayAll);

    render(withWrapper(<WorkshopSelection />));
    waitFor(() => {
      expect(mapInstance.map.getZoom).toHaveBeenCalled();
    });

    expect(autoFixClient.query.mock.calls[0][0].variables.payload.limit).toBe(
      0
    );
  });

  test("it should fetch all workshops when zoom level is above hide all level and bellow show all level", () => {
    jest.spyOn(mapInstance.map, "getZoom").mockImplementation(() => 9);

    render(withWrapper(<WorkshopSelection />));
    waitFor(() => {
      expect(mapInstance.map.getZoom).toHaveBeenCalled();
    });

    expect(autoFixClient.query.mock.calls[0][0].variables.payload.limit).toBe(
      Math.floor(9 * 1.5)
    );
  });

  test("Should call Map component with the right properties", async () => {
    const mapProperties = {
      ref: mapInstance,
      initalCordinates: [25.19741055892981, 55.27478409367688],
      googleMapKey: "AIzaSyBn3kw1bNdgmiXAczwr2DcKLAaW-M3nX14",
      woosMapKey: "woos-48c80350-88aa-333e-835a-07f4b658a9a4",
    };

    const { getByTestId } = render(withWrapper(<WorkshopSelection />));

    await waitFor(() => {
      expect(getByTestId(pageName)).not.toBeNull();
    });

    const MapComponent = jest.fn().mockImplementation((e: unknown) => e);
    jest.spyOn(designSystem, "Map").mockImplementation(MapComponent);

    expect(mapMockFN).toHaveBeenCalled();
    const mapComponentParameters = mapMockFN.mock.calls[0][0];

    expect(mapComponentParameters.initalCordinates).toStrictEqual(
      mapProperties.initalCordinates
    );
    expect(mapComponentParameters.googleMapKey).toStrictEqual(
      mapProperties.googleMapKey
    );
    expect(mapComponentParameters.woosMapKey).toStrictEqual(
      mapProperties.woosMapKey
    );
  });
});
