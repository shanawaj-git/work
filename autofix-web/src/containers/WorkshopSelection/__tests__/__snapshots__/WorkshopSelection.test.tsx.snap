// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkshopSelection /> Component Should render markers when state update 1`] = `"<div data-testid=\\"workshop.selection\\" class=\\"min-h-screen flex flex-col h-screen\\"><div data-testid=\\"PageHeader\\" class=\\"min-h-screen p-6 \\" style=\\"min-height: fit-content;\\"><div><button class=\\"root StyledButton-sc-1apep4b-0 bXswYh mb-5 min-w-[40px] min-h-[40px] w-[40px] h-[40px] MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"workshop.selection-back-button\\" style=\\"border-color: rgba(220, 220, 220, 0.2);\\"><div style=\\"visibility: visible;\\"><img src=\\"arrow-back-icon.svg\\" alt=\\"BackArrow\\"></div></button><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-xl whitespace-pre-wrap !mt-4 css-1uw3dvl-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Pick your workshop</span></p><div color=\\"#FCF7F7\\" class=\\"text-xl whitespace-pre-wrap !mt-4\\"></div></div><div class=\\"MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root\\" role=\\"combobox\\" data-testid=\\"AutoComplete\\" inputmode=\\"search\\"><div class=\\"flex w-full rounded relative\\"><div class=\\"z-10 flex justify-center items-center h-auto px-4 absolute top-0 left-0 bottom-0\\"><img src=\\"search-icon.svg\\" alt=\\"SearchIcon\\"></div><div class=\\"MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root sc-81b144b0-0 lcKTQR css-wb57ya-MuiFormControl-root-MuiTextField-root\\"><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" autocomplete=\\"off\\" id=\\"workshop-selection-autocomplete\\" placeholder=\\"Search for an address or workshop\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input\\" aria-autocomplete=\\"list\\" autocapitalize=\\"none\\" spellcheck=\\"false\\" value=\\"\\"><div class=\\"MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Clear\\" title=\\"Clear\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"CloseIcon\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Open\\" title=\\"Open\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"ArrowDropDownIcon\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></button></div><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-nnbavb\\"><span class=\\"notranslate\\">​</span></legend></fieldset></div></div></div></div></div></div><div class=\\"h-full\\"><div data-testid=\\"map\\"><div><div data-testid=\\"marker\\"></div><div data-testid=\\"marker\\"></div><div data-testid=\\"marker\\"></div><div data-testid=\\"marker\\"></div><div data-testid=\\"marker\\"></div></div></div></div></div>"`;

exports[`<WorkshopSelection /> Component should contain map component 1`] = `"<div data-testid=\\"workshop.selection\\" class=\\"min-h-screen flex flex-col h-screen\\"><div data-testid=\\"PageHeader\\" class=\\"min-h-screen p-6 \\" style=\\"min-height: fit-content;\\"><div><button class=\\"root StyledButton-sc-1apep4b-0 bXswYh mb-5 min-w-[40px] min-h-[40px] w-[40px] h-[40px] MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"workshop.selection-back-button\\" style=\\"border-color: rgba(220, 220, 220, 0.2);\\"><div style=\\"visibility: visible;\\"><img src=\\"arrow-back-icon.svg\\" alt=\\"BackArrow\\"></div></button><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-xl whitespace-pre-wrap !mt-4 css-1uw3dvl-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Pick your workshop</span></p><div color=\\"#FCF7F7\\" class=\\"text-xl whitespace-pre-wrap !mt-4\\"></div></div><div class=\\"MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root\\" role=\\"combobox\\" data-testid=\\"AutoComplete\\" inputmode=\\"search\\"><div class=\\"flex w-full rounded relative\\"><div class=\\"z-10 flex justify-center items-center h-auto px-4 absolute top-0 left-0 bottom-0\\"><img src=\\"search-icon.svg\\" alt=\\"SearchIcon\\"></div><div class=\\"MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root sc-81b144b0-0 lcKTQR css-wb57ya-MuiFormControl-root-MuiTextField-root\\"><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" autocomplete=\\"off\\" id=\\"workshop-selection-autocomplete\\" placeholder=\\"Search for an address or workshop\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input\\" aria-autocomplete=\\"list\\" autocapitalize=\\"none\\" spellcheck=\\"false\\" value=\\"\\"><div class=\\"MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Clear\\" title=\\"Clear\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"CloseIcon\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Open\\" title=\\"Open\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"ArrowDropDownIcon\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></button></div><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-nnbavb\\"><span class=\\"notranslate\\">​</span></legend></fieldset></div></div></div></div></div></div><div class=\\"h-full\\"><div data-testid=\\"map\\"><div><div data-testid=\\"marker\\"></div></div></div></div></div>"`;

exports[`<WorkshopSelection /> Component should render 1`] = `"<div data-testid=\\"workshop.selection\\" class=\\"min-h-screen flex flex-col h-screen\\"><div data-testid=\\"PageHeader\\" class=\\"min-h-screen p-6 \\" style=\\"min-height: fit-content;\\"><div><button class=\\"root StyledButton-sc-1apep4b-0 bXswYh mb-5 min-w-[40px] min-h-[40px] w-[40px] h-[40px] MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"workshop.selection-back-button\\" style=\\"border-color: rgba(220, 220, 220, 0.2);\\"><div style=\\"visibility: visible;\\"><img src=\\"arrow-back-icon.svg\\" alt=\\"BackArrow\\"></div></button><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-xl whitespace-pre-wrap !mt-4 css-1uw3dvl-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Pick your workshop</span></p><div color=\\"#FCF7F7\\" class=\\"text-xl whitespace-pre-wrap !mt-4\\"></div></div><div class=\\"MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root\\" role=\\"combobox\\" data-testid=\\"AutoComplete\\" inputmode=\\"search\\"><div class=\\"flex w-full rounded relative\\"><div class=\\"z-10 flex justify-center items-center h-auto px-4 absolute top-0 left-0 bottom-0\\"><img src=\\"search-icon.svg\\" alt=\\"SearchIcon\\"></div><div class=\\"MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root sc-81b144b0-0 lcKTQR css-wb57ya-MuiFormControl-root-MuiTextField-root\\"><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" autocomplete=\\"off\\" id=\\"workshop-selection-autocomplete\\" placeholder=\\"Search for an address or workshop\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input\\" aria-autocomplete=\\"list\\" autocapitalize=\\"none\\" spellcheck=\\"false\\" value=\\"\\"><div class=\\"MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Clear\\" title=\\"Clear\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"CloseIcon\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Open\\" title=\\"Open\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"ArrowDropDownIcon\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></button></div><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-nnbavb\\"><span class=\\"notranslate\\">​</span></legend></fieldset></div></div></div></div></div></div><div class=\\"h-full\\"><div data-testid=\\"map\\"><div><div data-testid=\\"marker\\"></div></div></div></div></div>"`;
